### DISCUSSION TITLE
Is it Possible to Add Disk to HA in Proxmox and Access through Media Folder?

### DISCUSSION THREAD

**Discussion by Photo64** [tteck/Proxmox#812] [Q&A] [answered]
I was attempting to mount a folder to the /media folder within HA that is currently in a Proxmox VM via a Ubuntu LXC container to store media files. Unfortunately while I could access the files via my mac and linux laptops I couldn't get the files to mount within HA. I did find some documentation about squashing and unsquashing folders it was a bit above my understanding. 

My next thought was maybe I could just mount a drive within the HA VM and be able to mount the folder easier that way within the HA /media folder. Would anyone know if this is possible or will I still run into the same issue when trying to mount it from a different lxc? I was reluctant to just load a bunch of media files within the /media folder of HA taking up valuable space and was trying to find a way to offload them and just have a mounted drive to them.

**Comment by tteck** [comment, marked as answer]
You can always increase the HAOS disk size.

![Screenshot 2022-12-07 9 51 57 PM](https://user-images.githubusercontent.com/21342670/206344650-d7a0a54e-845c-4d37-b79b-88f2114682de.png)

**Comment by Photo64** [comment]
I suspect I'll have to go this route as there doesn't seem to be an easy method of mounting an external folder. Thx

**Comment by dmshimself** [comment]
Before I moved to using an LXC container which makes this easy, I used to run a HASSOS service which was basically a shell script to do an NFS mount into the right place and then called that as part of an automation trigger by home assistant starting up.  That might be worth a try

**Comment by Photo64** [comment]
@dmshimself  Would you have an example of that? I tried doing something similar but always got a return code of 255 even though I had all the permissions set correctly and I could actually access it via my mac laptop and linux laptops, just couldn't through HAOS in Proxmox. Then started reading about squashing and squashing folders, started getting over my head. If you have a workaround I would love to try it.

**Reply by dmshimself** [reply]
Sure thing.   You will need the ssh addon to get this mount to work.  I initially used the ssh addon to test as well, so having started and used the shh addon, I then manually ran 'stuff' until the following worked for me.

mount -t nfs4 10.13.0.2:/OurData/OurData /media/ourdata

I chose to create this in a folder that didn't exist as standard, so I needed to do a mkdir.  Then I ran 'ls' before and afterwards to check.

I do recall getting an error 255 as well.  In my case it was because my NFS share on the server was an NFS4 one and I was using just mount -t nfs .  But testing it by running the command manually showed the error and allowed me to correct it.  In my case the nfs4 share allows everyone read access.

The automation once all was well manually, was:

- id: '1657605950887'
  alias: Mount Images at boot
  description: Mount ourdata folder at boot.
  trigger:
  - platform: homeassistant
    event: start
  condition: []
  action:
  - delay:
      hours: 0
      minutes: 1
      seconds: 0
      milliseconds: 0
  - service: hassio.addon_stdin
    data:
      addon: a0d7b954_ssh
      input: mount -t nfs4 10.13.0.2:/OurData/OurData /media/ourdata
  mode: single

I found I needed a delay so that the hassos system could settled.  There is probably a more elegant way than just a simple delay, but it worked for me.

Now that I have moved to LXC containers, it's just a question of adding a bindmount into the right place in the lxc config file.  In case you were thinking of moving to that, I have something like...

mp1: /OurData,mp=/OurData,ro=1
mp2: /OurData/OurData,mp=/OurData/OurData,ro=1

in the relevant <nnn>.conf file

I found that the lxc approach gave huge resource benefits and as I only have a small server, it was a great move

**Reply by Photo64** [reply]
So some of the differences I see from your example and my setup (below) are the use of **nfs4** versus cifs and -o. The automation looks somewhat similar except for the delay. I tried many approaches but fairly sure I didn't try the use of nfs4. It just boggles me because I have no issues accessing this folder with something other than HAOS in Proxmox.

Just to confirm, my HAOS is in a Proxmox VM (curtesy of tteck's script) and the folder I am trying to mount is in a ubuntu lxc. Is this what you are currently doing to mount your folder?


`shell_command:
  mount_music_folder: mkdir -p /media/sambashare;mount -t cifs -o vers=1.0,noserverino,username=anonymous,domain=WORKGROUP "//192.168.1.70/sambashare" /media/sambashare`

automation:
`- alias: Mount Sambashare Music Folder
  id: xxxxxxxxx
  description: Mounts Samba share for use with Music Assistant 
  trigger:
  - platform: homeassistant
    event: start
  condition: []
  action:
  - service: shell_command.mount_music_folder
    data: {}`

**Comment by dmshimself** [comment]
My nfs server is an LXC too, just debian instead.  I couldn't get the shell method to work at all, so I used the method I have laid out.  If you start with the ssh addon and manually try, then I think you will have more luck in diagnosing the issue.

**Reply by Photo64** [reply]
So admittedly I am more of a novice around this but always trying to learn. I have the SSH&Terminal addon in HA and at the command line I entered "mount -t nfs4 192.168.1.70:/media /media/sambashare" however it still returns a "Permission Denied". 

Also the bindmounts you added I assume are in the <lxc>.conf file. Would I need to add this prior to trying the mount command?

**Comment by dmshimself** [comment]
You will need to use a samba connection string on the mount if your share is a samba one and an NFS mount if it is an NFS share.  They are not interchangable.  To go back slightly one stage, do you have a linux box that you can use to test mounting?  As you are using proxmox, you could easily spin up an ubuntu VM and use that to test.  The HASSOS VM you have is running linux, but without too much in the way of a user interface.  It might be worth getting a standard mount command working on an easy platform first and then use the ssh into your HA VM.

Important note - the binds I gave are only to be used if you are using a ha LXC, not a VM, they will not work on a VM

**Reply by Photo64** [reply]
I have no issues with using samba or ssh from either my mac laptop or my ubuntu laptop. With both I can mount the media folder from the ubuntu lxc container. 
I can not mount the ubuntu lxc folder in the HA VM. Every time I try I get a "Permission Denied" error when trying to mount the folder when using the SSH & Terminal addon within HA.  Using nfs or nfs4 or cifs, they return the same error.  What I don't know is if the Permission Denied error is on the lxc container or the HAOS VM. 

Thanks for your suggestions but it seems it just can't be done and I'll likely have to go the route that tteck suggested above.

**Comment by dmshimself** [comment]
Sure thing - it's a hard error to work through as you'd need to go 
through the log files on the ubuntu container running as your file 
server.  I share your pain!  I guess you might come back to it 
sometime.  Anyway good luck and I'm glad you have an alternative 
approach that gives you what you need.
